scripts/images/improveImages.mjs

```diff
@@ -0,0 +1,92 @@
+import fs from 'fs'
+import find from 'find'
+import filesize from 'filesize'
+import imagemin from 'imagemin'
+import imageminGifsicle from 'imagemin-gifsicle'
+import imageminJpegtran from 'imagemin-jpegtran'
+import imageminOptipng from 'imagemin-optipng'
+import imageminSvgo from 'imagemin-svgo'
+import parseFilepath from 'parse-filepath'
+import chalk from 'chalk'
+
+const plugins = [
+  imageminGifsicle({}),
+  imageminJpegtran({}),
+  imageminOptipng({}),
+  imageminSvgo({})
+]
+
+let savedSize = 0
+
+const run = async () => {
+  const regex = new RegExp(/\.gif|\.jpeg|\.jpg|\.png$/)
+
+  const files = find.fileSync(regex, '.')
+
+  const filteredFiles = files.filter(file => !file.match(/node_modules\//))
+
+  for (const file of filteredFiles) {
+    await optimized(file)
+  }
```

Comment:
Ideally this should be ignoring anything in the `.gitignore` file (not just `node_modules`). There may be a lot of temp/build sub-directories for packages like DevTools that should also be skipped as far as this script is concerned.

But that's probably too heavy handed to build. Maybe instead we could just re-purpose this broad/generic script to be very DevTools specific? In other words:

```suggestion
  const files = find.fileSync(regex, 'packages/react-devtools-extensions/icons');

  for (const file of files) {
    await optimized(file)
  }
```

This also has a side effect of getting this script down from ~40 seconds runtime to ~5 seconds.

Reply:
I made it general because there are some images outside the devtools extensions folder, e.g.: `packages/react-devtools/icons/icon128.png` and `scripts/bench/benchmarks/hacker-news/logo.png` (see more in example output).
Should we ignore it?

Reply:
Understood, but nothing in the <scripts> directory gets deployed so we don't really need to care about those. I think there's only one image outside of `packages/react-devtools-extensions` that got optimized.

Reply:
Actually 2: `packages/react-devtools/icons/icon128.png` and `scripts/bench/benchmarks/hacker-news/logo.png`, but yeah, cannot compared with `packages/react-devtools-extensions`. 

Will update to only DevTools.

Reply:
Right but nothing in `scripts` ships or gets deployed (so that doesn't matter).